server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RecipientRewriteTableProcessor.java

```diff
@@ -106,14 +107,26 @@ public RrtExecutionResult(ImmutableList<MailAddress> newRecipients, ImmutableLis
     private final MailetContext mailetContext;
 
     private static final Function<Mapping, Optional<MailAddress>> mailAddressFromMapping =
-        addressMapping -> {
-            try {
-                return Optional.of(new MailAddress(addressMapping.asString()));
-            } catch (AddressException e) {
-                return Optional.empty();
+        mapping -> {
+            Type type = mapping.getType();
+            switch (type) {
+                case Address:
+                    return parseMappingToMailAddress(mapping.asString());
```

Comment:
Why don't we use `Mapping::getAddress` ?

Why don't Mapping::getAddress don't directly return an `Optional<MailAddress>` ?